### PR TITLE
Add MySQL ALPN proto to 'tsh proxy db'

### PR DESCRIPTION
**Pull request by jakule**

### CODE REVIEW

File: tool/tsh/proxy.go [hunk 1]
```diff
@@ -186,14 +186,28 @@ func onProxyCommandDB(cf *CLIConf) error {
...
-	})
+	}
+
+	if database.Protocol == defaults.ProtocolMySQL {
+		database, err := getDatabase(cf, client, database.ServiceName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		mysqlSeverVersionProto := mySQLVersionToProto(database)
```
Comment (r0mant) [inline tool/tsh/proxy.go:204]:
```suggestion
		mysqlServerVersionProto := mySQLVersionToProto(database)
```

File: tool/tsh/proxy.go [hunk 2]
```diff
@@ -186,14 +186,28 @@ func onProxyCommandDB(cf *CLIConf) error {
...
+		if mysqlSeverVersionProto != "" {
+			proxyOpts.protocols = append(proxyOpts.protocols, alpncommon.Protocol(mysqlSeverVersionProto))
```
Comment (r0mant) [inline tool/tsh/proxy.go:206]: Why are we appending it instead of just setting?

File: tool/tsh/proxy.go [hunk 3]
```diff
@@ -186,14 +186,28 @@ func onProxyCommandDB(cf *CLIConf) error {
...
 		proxyAddr: client.WebProxyAddr,
 		protocols: []alpncommon.Protocol{alpncommon.Protocol(database.Protocol)},
 		listener:  listener,
 		insecure:  cf.InsecureSkipVerify,
 		certFile:  certFile,
 		keyFile:   keyFile,
-	})
+	}
+
+	if database.Protocol == defaults.ProtocolMySQL {
```
Comment (r0mant) [inline tool/tsh/proxy.go:198]: I would encapsulate this logic into a separate method e.g. `getDatabaseALPNProcotol` or something like this.

File: tool/tsh/proxy.go (hunk 2 above)
Comment (jakule) [reply tool/tsh/proxy.go:206]: We still need the "old" MySQL protocol field for backward compatibility and some validation in DB proxy.

File: lib/srv/db/server.go [hunk 4]
```diff
@@ -318,7 +318,7 @@ func (s *Server) startDatabase(ctx context.Context, database types.Database) err
 	}
 	if err := fetchMySQLVersion(ctx, database); err != nil {
 		// Log, but do not fail. We will fetch the version later.
-		s.log.Warnf("Failed to fetch the MySQL version for %s", database.GetName())
+		s.log.Warnf("Failed to fetch the MySQL version for %s: error: %v", database.GetName(), err)
```
Comment (marcoandredinis) [inline lib/srv/db/server.go:321]: nit
```suggestion
		s.log.Warnf("Failed to fetch the MySQL version for %s: %v", database.GetName(), err)
```

File: tool/tsh/db.go [hunk 5]
```diff
@@ -432,6 +417,64 @@ func maybeStartLocalProxy(cf *CLIConf, tc *client.TeleportClient, profile *clien
...
+			if err != nil {
+				return localProxyOpts{}, trace.Wrap(err)
+			}
+		}
+
+		mysqlServerVersionProto := mySQLVersionToProto(arg.database)
+		if mysqlServerVersionProto != "" {
+			opts.protocols = append(opts.protocols, common.Protocol(mysqlServerVersionProto))
+		}
+	}
```
Comment (marcoandredinis) [inline tool/tsh/db.go:473]: I think we are getting to a point where we should extract those methods closer to the DB logic we have for each Protocol
Maybe not with this PR, but if we add another `if ... protocolX` I thnik we should re-write this flow

**Review by marcoandredinis — APPROVED** [review]
LGTM
Just a small nit

**Review by r0mant — APPROVED** [review]

**Pull request merged**